ipaserver/plugins/service.py

```diff
@@ -289,7 +289,7 @@ def check_required_principal(ldap, principal):
     except errors.ValidationError:
         service_types = ['HTTP', 'ldap', 'DNS', 'dogtagldap']
         if principal.service_name in service_types:
-            raise errors.ValidationError(name='principal', error=_('This principal is required by the IPA master'))
+            raise errors.ValidationError(name=principal, error=_('This principal is required by the IPA master'))
```

Comment:
name is the attribute or option that failed validation. If you want to include the principal being rejected then include it in the error message.